sources/packages/backend/apps/api/src/route-controllers/application/application.aest.controller.ts

```diff
@@ -68,15 +68,36 @@ export class ApplicationAESTController extends BaseController {
    * @returns Application details
    */
   @Get(":applicationId")
-  @ApiNotFoundResponse({ description: "Application not found." })
+  @ApiNotFoundResponse({
+    description:
+      "Application not found or current application for provided parent application not found.",
+  })
   async getApplication(
     @Param("applicationId", ParseIntPipe) applicationId: number,
     @Query("loadDynamicData", new DefaultValuePipe(true), ParseBoolPipe)
     loadDynamicData: boolean,
+    @Query("isParentApplication", new DefaultValuePipe(false), ParseBoolPipe)
+    isParentApplication: boolean,
   ): Promise<ApplicationSupplementalDataAPIOutDTO> {
+    let currentApplicationId = applicationId;
+    if (isParentApplication) {
+      currentApplicationId =
+        await this.applicationService.getApplicationIdByParentApplicationId(
+          applicationId,
+        );
+      if (!currentApplicationId) {
+        throw new NotFoundException(
+          `Current application for application ${applicationId} was not found.`,
```

Comment:
`Current application for application for provided parent application ${applicationId} was not found.`